Mvp/create contact

GameyMonster opened the pull request:
Create 3 different class: Contacts, CreateContacts, and ListOfContacts
Contacts: Create a serializable for the Contact: name and phone
CreateContacts: User can able to insert their name and phone, and should able to interact the ListView 
ListOfContacts: It's part of CreateContacts.kt when interact the ListView; still working on the code for the Call/Text

GameyMonster committed:
Create a 2 types of class: Create Contact and Display their name and phone number

GameyMonster committed:
Fix some stuff

GameyMonster committed:
Merge branch 'master' into mvp/CreateContact

GameyMonster committed:
Create a bigger size of the ListView

GameyMonster committed:
Merge remote-tracking branch 'origin/mvp/CreateContact' into mvp/CreateContact

GameyMonster committed:
Create a bigger size of the frame for the ListView

GameyMonster committed:
Change some code

GameyMonster committed:
Display the contact Name and Phone Number

GameyMonster committed:
Edit the code for ListOfContacts, and create some comments

GameyMonster committed:
Create the length for the name and phone number and fix the activity for the ListOfContacts

kouYang2015 commented on app/build.gradle:
```diff
@@ -2,6 +2,8 @@ import org.apache.tools.ant.taskdefs.condition.Os
 
 plugins {
     id 'com.android.application'
+    id 'kotlin-android'
+    id 'kotlin-android-extensions'
```
Just wondering what this was used for?

GameyMonster replied:
I'll delete this I was testing something for my MVP, but it didn't need it since it work without it

kouYang2015 commented on app/src/main/res/drawable/ic_baseline_call.xml:
```diff
@@ -0,0 +1,5 @@
+<vector android:height="24dp" android:tint="#000000"
+    android:viewportHeight="24" android:viewportWidth="24"
+    android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
+    <path android:fillColor="@android:color/white" android:pathData="M20.01,15.38c-1.23,0 -2.42,-0.2 -3.53,-0.56 -0.35,-0.12 -0.74,-0.03 -1.01,0.24l-1.57,1.97c-2.83,-1.35 -5.48,-3.9 -6.89,-6.83l1.95,-1.66c0.27,-0.28 0.35,-0.67 0.24,-1.02 -0.37,-1.11 -0.56,-2.3 -0.56,-3.53 0,-0.54 -0.45,-0.99 -0.99,-0.99H4.19C3.65,3 3,3.24 3,3.99 3,13.28 10.73,21 20.01,21c0.71,0 0.99,-0.63 0.99,-1.18v-3.45c0,-0.54 ...
+</vector>
```
Just wondering what this was

GameyMonster replied:
Image of the Phone

kouYang2015 commented on app/src/main/res/drawable/ic_baseline_message.xml:
```diff
@@ -0,0 +1,5 @@
+<vector android:autoMirrored="true" android:height="24dp"
+    android:tint="#000000" android:viewportHeight="24"
+    android:viewportWidth="24" android:width="24dp" xmlns:android="http://schemas.android.com/apk/res/android">
+    <path android:fillColor="@android:color/white" android:pathData="M20,2L4,2c-1.1,0 -1.99,0.9 -1.99,2L2,22l4,-4h14c1.1,0 2,-0.9 2,-2L22,4c0,-1.1 -0.9,-2 -2,-2zM18,14L6,14v-2h12v2zM18,11L6,11L6,9h12v2zM18,8L6,8L6,6h12v2z"/>
+</vector>
```
Wondering what this is too.

GameyMonster replied:
Image of a message

kouYang2015 commented on app/src/main/res/values/strings.xml:
```diff
@@ -29,4 +29,18 @@
     <string name="coming_soon_text">Coming Soon</string>
     <string name="onSpeech">Turn On</string>
     <string name="offSpeech">Turn Off</string>
+    <string name="first_name">First Name</string>
+    <string name="last_name">Last Name</string>
+    <string name="phone_number">Phone Number</string>
+    <string name="btnsave">Save</string>
+    <string name="add_new_contacts">Add New Contacts</string>
+    <string name="btn_new">New</string>
+    <string name="btn_delete">Delete</string>
+    <string name="load">Load</string>
+    <string name="btnadd">Add</string>
+    <string name="textUserview">TextView</string>
+    <string name="textPhoneview">TextView</string>
+    <string name="back">Back</string>
+    <string name="phone_message">TODO</string>
+    <string name="phone_call">phone_call</string>
```
Is there a way that we can use some of the older names and ids so that we don't have duplicates? Just take another look.

GameyMonster replied:
I'll delete some of the string that doesn't required my MVP

kouYang2015 commented on app/src/main/java/com/metrostateics499/vre_app/CreateContacts.kt:
```diff
@@ -0,0 +1,109 @@
+        setContentView(R.layout.activity_create_contacts)
+
+        btnNewAccount = findViewById(R.id.btnAddContact)
+        fullName = findViewById(R.id.editTextPersonName)
+        phoneNum = findViewById(R.id.editTextPhone)
+        listContact = findViewById(R.id.contactList)
+        btnDelete = findViewById(R.id.btnDeleteContact)
+
+        listNames = ArrayList()
+        listOfContact = ArrayList()
+        createAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, listNames)
+
+        listContact.adapter = createAdapter
+
+        btnNewAccount.setOnClickListener {
+            if (fullName.text.toString().isNotEmpty() && phoneNum.text.toString().isNotEmpty()) {
+                if (fullName.length() in 6..36 ) {
```
let's change minimum from 6 to 2. User might want to only do first name and the shortest is 2 letters.
```suggestion
        fullName = findViewById(R.id.editTextPersonName)
        phoneNum = findViewById(R.id.editTextPhone)
        listContact = findViewById(R.id.contactList)
        btnDelete = findViewById(R.id.btnDeleteContact)

        listNames = ArrayList()
        listOfContact = ArrayList()
        createAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, listNames)

        listContact.adapter = createAdapter

        btnNewAccount.setOnClickListener {
            if (fullName.text.toString().isNotEmpty() && phoneNum.text.toString().isNotEmpty()) {
                if (fullName.length() in 2..36 ) {
```

kouYang2015 replied:
```suggestion
                if (fullName.length() in 2..36 ) {
```

kouYang2015 commented on app/src/main/java/com/metrostateics499/vre_app/CreateContacts.kt:
```diff
@@ -0,0 +1,109 @@
+
+        btnNewAccount.setOnClickListener {
+            if (fullName.text.toString().isNotEmpty() && phoneNum.text.toString().isNotEmpty()) {
+                if (fullName.length() in 6..36 ) {
```
```suggestion
                if (fullName.length() in 2..36 ) {
```

I think we should change minimum to 2 since some Users might want to only put acronym or first names only.

GameyMonster replied:
Okay.

kouYang2015 reviewed (changes requested):
Mostly look good. Take a look at some of my comments

GameyMonster committed:
Delete some of the strings.xml, and edit some current string reference

GameyMonster committed:
Add ImageView to Interact: can able to call or text

GameyMonster committed:
Remove the build.gradle : unnecessary needs
ListOfContacts: Create use the Image for the ListOfContacts.kt, either to call nor text
And edit some comments to the CreateContacts

GameyMonster committed:
Additional Edit from the gradle
